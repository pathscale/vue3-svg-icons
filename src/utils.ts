import qs from "query-string";

import { Query } from "./types";

export function parseQuery(id: string): Query {
  const [filename, query] = id.split("?", 2);
  if (!query) return { vue: false };

  const raw = qs.parse(query);
  if (!("vue" in raw)) return { vue: false };

  return {
    ...raw,
    filename,
    vue: true,
    index: raw.index && Number(raw.index),
    src: "src" in raw,
    scoped: "scoped" in raw,
  } as Query;
}
