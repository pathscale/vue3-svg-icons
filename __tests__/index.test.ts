import { write } from "./utils";

test("basic", async () => {
  const res = await write({
    input: "basic/index.js",
    outDir: "basic",
    options: {},
  });

  for (const f of await res.js()) expect(f).toMatchSnapshot("js");
  for (const f of await res.svg()) expect(f).toMatchSnapshot("svg");

  // TODO: Switch back to truthy after adding serialization
  await expect(res.isSvg()).resolves.toBeFalsy();
});
