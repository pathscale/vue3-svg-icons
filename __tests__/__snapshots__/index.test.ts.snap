// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic: js 1`] = `
"import { resolveComponent, openBlock, createBlock, createApp } from 'vue';

var script = {};

function render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_vue3_svg = resolveComponent(\\"vue3-svg\\");

  return (openBlock(), createBlock(_component_vue3_svg, { \\"500px.svg\\": \\"\\" }))
}

script.render = render;
script.__file = \\"__tests__/fixtures/basic/App.vue\\";

createApp(script).mount(\\"body\\");
"
`;
